Source/Public/Publish-Bicep.ps1

```diff
@@ -1,14 +1,30 @@
 function Publish-Bicep {
     [CmdletBinding()]
     param (
-        [Parameter(Mandatory, Position = 1)]
+        [Parameter(ParameterSetName = 'Default', Mandatory, Position = 1)]
+        [Parameter(ParameterSetName = 'Registry', Position = 1)]
         [ValidateNotNullOrEmpty()]
-        [ValidateScript({Test-Path $_})]
+        [ValidateScript( { Test-Path $_ })]
         [string]$Path,
 
-        [Parameter(Mandatory, Position = 2)]
+        [Parameter(ParameterSetName = 'Default', Position = 2)]
         [ValidateNotNullOrEmpty()]
-        [string]$Target          
+        [string]$Target,
+        
+        [Parameter(ParameterSetName = 'Registry', Mandatory, Position = 2)]
```

Comment:
It would be good if we could have some kind of input validation similar to $target in pr #239 .  This would prevent anyone from inputing bad data that will throw errors later.